grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ThingImpl.java

```diff
@@ -70,13 +70,13 @@
  *           For example {@link ai.grakn.concept.EntityType} or {@link RelationshipType}
  */
 public abstract class ThingImpl<T extends Thing, V extends Type> extends ConceptImpl implements Thing {
-    private final Cache<Label> cachedInternalType = new Cache<>(Cacheable.label(), () -> {
+    private final Cache<Label> cachedInternalType = Cache.createTxCache(this, Cacheable.label(), () -> {
         int typeId = vertex().property(Schema.VertexProperty.THING_TYPE_LABEL_ID);
         Optional<Type> type = vertex().tx().getConcept(Schema.VertexProperty.LABEL_ID, typeId);
         return type.orElseThrow(() -> GraknTxOperationException.missingType(getId())).getLabel();
     });
 
-    private final Cache<V> cachedType = new Cache<>(Cacheable.concept(), () -> {
+    private final Cache<V> cachedType = Cache.createTxCache(Cacheable.concept(), () -> {
```

Comment:
I think this one needs to be passed `this` so it is cleared.

It might be a good idea to rename the `Cache#createTxCache` method that doesn't take an owner, otherwise we will definitely make this mistake again :P